src/Element/OgAutocomplete.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace Drupal\og\Element;
+
+use Drupal\Component\Utility\Crypt;
+use Drupal\Core\Entity\Element\EntityAutocomplete;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Site\Settings;
+
+/**
+ * Provides an entity autocomplete with OG group context.
+ *
+ * The #default_value accepted by this element is either an entity object or an
+ * array of entity objects.
+ *
+ * @FormElement("og_autocomplete")
+ */
+class OgAutocomplete extends EntityAutocomplete {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function processEntityAutocomplete(array &$element, FormStateInterface $form_state, array &$complete_form) {
+    // Nothing to do if there is no target entity type.
+    if (empty($element['#target_type'])) {
+      throw new \InvalidArgumentException('Missing required #target_type parameter.');
+    }
+
+    // Provide default values and sanity checks for the #autocreate parameter.
+    if ($element['#autocreate']) {
+      if (!isset($element['#autocreate']['bundle'])) {
+        throw new \InvalidArgumentException("Missing required #autocreate['bundle'] parameter.");
+      }
+      // Default the autocreate user ID to the current user.
+      $element['#autocreate']['uid'] = isset($element['#autocreate']['uid']) ? $element['#autocreate']['uid'] : \Drupal::currentUser()->id();
```

Comment:
We need to inject the current_user service.

Reply:
We are in a static method: ```public static function processEntityAutocomplete()``` so we can't use DI